File: services/customers-service/src/main/java/br/com/deltaglobalbank/customers/domain/bankAccount/BankAccountPurpose.java
package br.com.deltaglobalbank.customers.domain.bankAccount;

import java.util.Arrays;
import java.util.Locale;

public enum BankAccountPurpose {
    DISBURSEMENT,
    PAYOFF;

    public String toDatabaseValue() {
        return name().toLowerCase(Locale.ROOT);
    }

    public static BankAccountPurpose fromDatabaseValue(String value) {
        return Arrays.stream(values())
            .filter(purpose -> purpose.name().equalsIgnoreCase(value))
            .findFirst()
            .orElseThrow(() -> new IllegalArgumentException("bank account purpose invalid: " + value));
    }
}

File: services/customers-service/src/main/kotlin/br/com/deltaglobalbank/customers/domain/bankAccount/BankAccountPurpose.kt
package br.com.deltaglobalbank.customers.domain.bankAccount

enum class BankAccountPurpose {
    DISBURSEMENT,
    PAYOFF;

    fun toDatabaseValue(): String = name.lowercase()

    companion object {
        fun fromDatabaseValue(value: String): BankAccountPurpose =
            entries.firstOrNull { it.name.equals(value, ignoreCase = true) }
                ?: throw IllegalArgumentException("bank account purpose invalid: $value")
    }
}